lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -272,62 +312,31 @@ private static AblyException checkFor(Throwable t) {
 
 			/* send request body */
 			if(requestBody != null) {
-				conn.setDoOutput(true);
-				byte[] body = requestBody.getEncoded();
-				int length = body.length;
-				conn.setFixedLengthStreamingMode(length);
-				conn.setRequestProperty(CONTENT_TYPE, requestBody.getContentType());
-				conn.setRequestProperty(CONTENT_LENGTH, Integer.toString(length));
-				OutputStream os = conn.getOutputStream();
-				os.write(body);
+				writeRequestBody(requestBody, conn);
 			}
 
-			/* get response */
-			statusCode = conn.getResponseCode();
-			statusLine = conn.getResponseMessage();
-			if(statusCode != HttpURLConnection.HTTP_NO_CONTENT) {
-				contentType = conn.getContentType();
-				int contentLength = conn.getContentLength();
-				int successStatusCode = (method == POST) ? HttpURLConnection.HTTP_CREATED : HttpURLConnection.HTTP_OK;
-				is = (statusCode == successStatusCode) ? conn.getInputStream() : conn.getErrorStream();
-				if(is != null) {
-					int read;
-					if(contentLength == -1) {
-						ByteArrayOutputStream baos = new ByteArrayOutputStream();
-						byte[] buf = new byte[4 * 1024];
-						while((read = is.read(buf)) > -1) {
-							baos.write(buf, 0, read);
-						}
-						responseBody = baos.toByteArray();
-					} else {
-						int idx = 0;
-						responseBody = new byte[contentLength];
-						while((read = is.read(responseBody,  idx, contentLength - idx)) > -1) {
-							idx += read;
-						}
-					}
-				}
-			}
+			response = readResponse(conn);
 		} catch(IOException ioe) {
 			throw HostFailedException.checkFor(ioe);
 		} finally {
-			try {
-				if(is != null)
-					is.close();
-				if(conn != null)
-					conn.disconnect();
-			} catch(IOException ioe) {}
+			if(conn != null)
+				conn.disconnect();
 		}
 
-		if(statusCode == 0) {
+		if (response.statusCode == 0) {
```

Comment:
@gokhanbarisaker are you not going to address this?  If not, can you raise a separate issue for this problem so that it is fixed?